azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/ExtensionProfile.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import com.google.auto.value.AutoValue;
+import org.jclouds.json.SerializedNames;
+
+import java.util.List;
+
+@AutoValue
+public abstract class ExtensionProfile {
+
+    /**
+     * The Extensions of the extension profile
+     */
+    public abstract List<Extension> extensions();
+
+
+    @SerializedNames({"extensions"})
+    public static ExtensionProfile create(final List<Extension> extensions) {
+        return new AutoValue_ExtensionProfile(extensions);
```

Comment:
Enforce immutability. Also, as the list not declared nullable change to something like:
```java
return new AutoValue_ExtensionProfile(extensions == null ? ImmutableList.of() : ImmutableList.copyOf(extensions));
```

Also review the entire change set to enforce immutability in all collection properties when present.